roscala/src/main/scala/coop/rchain/rosette/Queue.scala

```diff
@@ -0,0 +1,101 @@
+package coop.rchain.rosette
+
+case class Queue(elems: Tuple) extends Ob {
+
+  def head: Ob = elems.elem.head
+
+  def tail: Ob = elems.elem.last
+
+  def nElems: Ob = Fixnum(elems.elem.length)
+
+  def depth(): Int =
+    elems.elem.length
+
+  def isEmpty(): Boolean =
+    (nElems.asInstanceOf[Fixnum] == Fixnum(0)).value
+
+  def enqueue(ele: Ob): Queue =
+    Queue(Tuple(elems.elem :+ ele))
+
+  def dequeue(): Option[Queue] =
+    if (isEmpty()) {
+      None
+    } else {
+      val v = elems.elem.headOption
+      Some(Queue(Tuple(elems.elem.drop(1))))
+    }
+
+  def reset(): Queue =
+    Queue.create()
+
+  override def indexedSize: Ob = nElems
+
+  override def nth(i: Int): Option[Ob] =
+    elems.elem.lift(i)
```

Comment:
The error handling here is different than the C++, but I'm inclined to leave it your way. because constraining the index to the queue by using the mod of the size is just weird and indeterministic.

In this case, let's have the primitives return 'PrimError(IndexOutOfBounds(i, size)))'

Reply:
The latest recommendation from the team is to implement it the same as the original C++ version for now. In other words, change this to use the same index mod queue size method to constrain the index.

Reply:
I think use the lift method of scala here is more elegant and concise, it's no need to do a index constraint.